utils/generator/templates/enum_template.java

```diff
@@ -0,0 +1,126 @@
+{% extends "base_template.java" %}
+{% block body %}
+public enum {{class_name}} {
+    {%- for param in params %}
+    {%- if param.description is defined or param.since is defined %}
+    /**
+     {%- if param.description is defined %}
+     {%- for d in param.description %}
+     * {{d}}
+     {%- endfor %}{% endif -%}
+     {%- if param.description is defined and (param.since is defined or param.see is defined) %}
+     *
+     {%- endif %}
+     {%- if param.since is defined %}
+     * @since SmartDeviceLink {{param.since}}
+     {%- endif %}
+     {%- if param.see is defined %}
+     * @see {{param.see}}
+     {%- endif %}
+     */
+    {%- endif %}
+    {%- if param.deprecated is defined %}
+    @Deprecated
+    {%- endif %}
+    {%- if kind == "simple" %}
+    {{param.name}}{{ "," if not loop.last }}
+    {%- elif kind == "complex" %}
+    {{param.name}}({{param.value}}){{ "," if not loop.last }}
+    {%- elif kind == "custom" %}
+    {{param.name}}({{param.internal}}){{ "," if not loop.last }}
+    {%- endif %}
+    {%- endfor %};
+
+    {%- if kind == "simple" %}
+
+    /**
+     * Convert String to {{class_name}}
+     *
+     * @param value String
+     * @return {{class_name}}
+     */
+    public static {{class_name}} valueForString(String value) {
+        try {
+            return valueOf(value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+    {%- elif kind == "complex" or kind == "custom" %}
+    {%- if return_type == "String" %}
+
+    private final String INTERNAL_NAME;
+
+    /**
+     * Private constructor
+     */
+    private {{class_name}}(String internalName) {
```

Comment:
and also rename this var to be `value`

Reply:
class_name renamed to value

Reply:
@o-mishch Thanks for making the change. I meant the `internalName` should be renamed to value. Sorry I did not mean the class name. Can we change the class name back and just change `internalName` to `value`. and `INTERNAL_NAME` to `VAULE`

Reply:
Like the following :
```
private {{class_name}}(String value) {
	this.VALUE = value;
}
```

Reply:
done

Reply:
Thanks